### PR TITLE
tsp support `@internal`

### PR DESCRIPTION
**Pull request by XiaofeiCao**
fix https://github.com/Azure/autorest.java/issues/2067

added test for cadl ranch

bump typespec-java to 0.6.1 for release

### CODE REVIEW

File: typespec-extension/src/code-model-builder.ts [hunk 1]
```diff
@@ -511,6 +512,12 @@ export class CodeModelBuilder {
     // check for generating protocol api or not
     codeModelOperation.generateProtocolApi = shouldGenerateProtocol(this.sdkContext, operation);
 
+    // currently we treat @Internal the same as @convenientAPI(false) and @protocolAPI(false) combined
+    if(isInternal(this.sdkContext, operation)) {
+      codeModelOperation.generateProtocolApi = false;
+      codeModelOperation.convenienceApi = undefined;
+    }
```
Comment (weidongxu-microsoft) [inline typespec-extension/src/code-model-builder.ts]: move it before https://github.com/Azure/autorest.java/blob/e53b5f2f1773b0c5dbefa971f62734589e32f986/typespec-extension/src/code-model-builder.ts#L453

other procedures already checking `convenienceApi` for schema tracking. E.g. 
https://github.com/Azure/autorest.java/blob/e53b5f2f1773b0c5dbefa971f62734589e32f986/typespec-extension/src/code-model-builder.ts#L720-L722

Comment (weidongxu-microsoft) [reply typespec-extension/src/code-model-builder.ts]: It means the model should not be generated, if only used in `@internal` operation.

File: typespec-tests/src/main/java/com/parameters/collectionformat/CollectionFormatClientBuilder.java
Comment (weidongxu-microsoft) [inline typespec-tests/src/main/java/com/parameters/collectionformat/CollectionFormatClientBuilder.java:1]: try merge latest main. I thought I've already regened these yesterday.

**Comment by weidongxu-microsoft** [comment]
also bump minor version
https://microsoft.sharepoint.com/teams/AzureDeveloperExperience/_layouts/OneNote.aspx?id=%2Fteams%2FAzureDeveloperExperience%2FShared%20Documents%2FShanghai%20Team%2FAzure%20SDK%20-%20Shanghai&wd=target%28Management%20SDK%2FAzure%20Java%20SDK.one%7C9A5CEBDF-9B6C-4131-97D8-AD20DB19B095%2FRelease%20of%20typespec-java%7C37A008C1-9100-4C62-AD1A-69AE1BC1ECE5%2F%29
onenote:https://microsoft.sharepoint.com/teams/AzureDeveloperExperience/Shared%20Documents/Shanghai%20Team/Azure%20SDK%20-%20Shanghai/Management%20SDK/Azure%20Java%20SDK.one#Release%20of%20typespec-java&section-id={9A5CEBDF-9B6C-4131-97D8-AD20DB19B095}&page-id={37A008C1-9100-4C62-AD1A-69AE1BC1ECE5}&end

**Review by weidongxu-microsoft — APPROVED** [review]

**Comment by weidongxu-microsoft** [comment]
LGTM

If possible, also try verify if this works good on the eventgrid tsp.

**Comment by XiaofeiCao** [comment]
I used their specs: https://github.com/Azure/azure-rest-api-specs/tree/feature/eventgrid/typespec/specification/eventgrid/Azure.Messaging.EventGrid

Here's the generated result: 
https://github.com/XiaofeiCao/autorest.java/blob/tsp_eventgrid_local_gen/typespec-tests/src/main/java/com/azure/messaging/eventgrid/EventGridAsyncClient.java

Seems no longer having issue as in the apiview.

**Review by haolingdong-msft — APPROVED** [review]
LGTM.

**Pull request merged**